### ISSUE TITLE
[BUG] Promtail ingestion does not work

### ISSUE THREAD

**Issue by debajyoti-truefoundry** [siglens/siglens#2019]
Logs from Siglens:
```
time="2024-12-30 10:26:13" level=error msg="sendErrorWithStatus: ProcessLokiApiIngestRequest at pkg/integrations/loki/loki.go:334: Unable to unmarshal request, err=invalid character '¶' looking for beginning of value"
time="2024-12-30 10:26:13" level=error msg="sendErrorWithStatus: ProcessLokiApiIngestRequest at pkg/integrations/loki/loki.go:334: Unable to unmarshal request, err=invalid character '\\u0092' looking for beginning of value"
time="2024-12-30 10:26:13" level=error msg="sendErrorWithStatus: ProcessLokiApiIngestRequest at pkg/integrations/loki/loki.go:334: Unable to unmarshal request, err=invalid character 'á' looking for beginning of value"
time="2024-12-30 10:26:13" level=error msg="sendErrorWithStatus: ProcessLokiApiIngestRequest at pkg/integrations/loki/loki.go:334: Unable to unmarshal request, err=invalid character 'Å' looking for beginning of value"
```

Logs from Promtail:
```
level=error ts=2024-12-30T10:26:41.742583776Z caller=client.go:430 component=client host=siglens-ingest-svc.siglens.svc.cluster.local:8081 msg="final error sending batch" status=400 tenant= error="server returned HTTP status 400 Bad Request (400): {\"error\":\"Unable to unmarshal request\"}"
level=error ts=2024-12-30T10:26:42.942795999Z caller=client.go:430 component=client host=siglens-ingest-svc.siglens.svc.cluster.local:8081 msg="final error sending batch" status=400 tenant= error="server returned HTTP status 400 Bad Request (400): {\"error\":\"Unable to unmarshal request\"}"
level=error ts=2024-12-30T10:26:44.242802542Z caller=client.go:430 component=client host=siglens-ingest-svc.siglens.svc.cluster.local:8081 msg="final error sending batch" status=400 tenant= error="server returned HTTP status 400 Bad Request (400): {\"error\":\"Unable to unmarshal request\"}"
```

Promtail config:
```yaml
config:
  clients:
    - url: http://loki:3100/loki/api/v1/push
    - url: http://siglens-ingest-svc.siglens.svc.cluster.local:8081/loki/api/v1/push
  snippets:
    extraRelabelConfigs:
      - regex: __meta_kubernetes_pod_label_(.+)
        action: labelmap
```

Promtail image: docker.io/grafana/promtail:2.9.10
Siglens image: siglens/siglens:1.0.7

**Comment by debajyoti-truefoundry** [comment]
I can confirm that this is fixed.

**Comment by debajyoti-truefoundry** [comment]
I can still see error logs like,
```
time="2024-12-31 10:25:40" level=error msg="sendErrorWithStatus: processPromtailLogs at pkg/integrations/loki/loki.go:284: Unable to convert line to string, err=<nil>"
```

@nkunal @AndrewHess

**Comment by AndrewHess** [comment]
> I can still see error logs like...

I've been looking into this, but can't reproduce this issue on promtail 2.9.10 and siglens 1.0.8 (since 1.0.8 has the fix that 1.0.7 was missing)
1. Does this happen for all logs, or just some of them?
2. Are you able to send me some example log lines that fail when promtail sends them to siglens?
3. Can you reproduce this running siglens/promtail as standalone images, or only inside a k8s cluster?
4. Can you share your full promtail config (or a minimal reproducible version that still has this issue)?
5. Is there anything else you can think of that will be helpful for me to reproduce/debug this?
